cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -161,26 +159,49 @@ private void downloadFileWithProgressBar(String url, Path target, HttpResponse<I
    *
    * @param source Path of file to copy
    * @param target Path of target directory
+   * @param test the flag about calling this method from test classes.
    */
-  private void copyFileWithProgressBar(Path source, Path target) throws IOException {
+  private void copyFileWithProgressBar(Path source, Path target, boolean test) throws IOException {
 
     try (InputStream in = new FileInputStream(source.toFile()); OutputStream out = new FileOutputStream(target.toFile())) {
+      long size;
+      if (test) {
+        size = 0L;
+      } else {
+        size = source.toFile().length();
+      }
```

Comment:
if you need to test this somehow, extract to protected method `long getFileSize(Path)` create a subclass of `FileAccessImpl` for test that overrides and returns `0` instead. Then in your JUnit replace the real `FileAccess` with the test instance... Seems a little bit more complicated but adding this `test` flag to our official API seems odd to me.

Reply:
I've implemented this as you've suggested, thanks.